libs/agno/agno/db/sqlite/sqlite.py

```diff
@@ -1974,6 +1995,322 @@ def rename_eval_run(
+        """
+        try:
+            table = self._get_table(table_type="traces", create_table_if_not_found=True)
+            if table is None:
+                return
+
+            with self.Session() as sess, sess.begin():
+                # First, check if trace exists and get current total_spans
+                existing = sess.execute(table.select().where(table.c.trace_id == trace.trace_id)).fetchone()
+
+                if existing:
+                    # Update with incremental values
+                    current_total = existing.total_spans
+                    current_errors = existing.error_count
+
+                    # Build update values - preserve non-null context fields
+                    update_values = {
+                        "total_spans": current_total + trace.total_spans,
+                        "error_count": current_errors + trace.error_count,
+                        "end_time_ns": trace.end_time_ns,
+                        "duration_ms": trace.duration_ms,
+                        "status": trace.status,
+                        # Update name only if new name looks like a root span (contains ".run" or ".arun")
+                        "name": trace.name if (".run" in trace.name or ".arun" in trace.name) else existing.name,
+                    }
+
+                    # Update context fields ONLY if new value is not None (preserve non-null values)
+                    if trace.run_id is not None:
+                        update_values["run_id"] = trace.run_id
+                    if trace.session_id is not None:
+                        update_values["session_id"] = trace.session_id
+                    if trace.user_id is not None:
+                        update_values["user_id"] = trace.user_id
+                    if trace.agent_id is not None:
+                        update_values["agent_id"] = trace.agent_id
+
+                    log_debug(
+                        f"  Updating trace with context: run_id={update_values.get('run_id', 'unchanged')}, "
+                        f"session_id={update_values.get('session_id', 'unchanged')}, "
+                        f"user_id={update_values.get('user_id', 'unchanged')}, "
+                        f"agent_id={update_values.get('agent_id', 'unchanged')}"
+                    )
+
+                    stmt = update(table).where(table.c.trace_id == trace.trace_id).values(**update_values)
+                    sess.execute(stmt)
+                else:
+                    # Insert new trace
+                    stmt = sqlite.insert(table).values(trace.to_dict())
+                    sess.execute(stmt)
+
+        except Exception as e:
+            log_error(f"Error creating trace: {e}")
+            # Don't raise - tracing should not break the main application flow
+
+    def get_trace(
+        self,
+        trace_id: Optional[str] = None,
+        run_id: Optional[str] = None,
+        session_id: Optional[str] = None,
```

Comment:
That doesn't make sense as a filter right? A trace_id would be unique? It should be a 1to1 with run_id, so that might make sense, but the others don't make sense.

Reply:
yeah ok that makes sense.